### ISSUE TITLE
Add an option for delete note

### ISSUE THREAD

**Issue by yashk2000** [kiwix/kiwix-android#1916]
**Is your feature request related to a problem? Please describe.**
Currently, there is a feature for save and edit notes. But there is not an option for deleting a note. If a user wants to erase a note, the user has to select all the text and erase it, but the file still stays in memory. 

There is a feature to delete **all** notes, but not to delete a **single** note. 

**Describe the solution you'd like**
Provide an option for deleting a note.

**Describe alternatives you've considered**
An option for deleting an existing note can be added to the menu of the note dialog.
It'll be made visible only after the user saves a new note, or if the user opens an existing note. 

**Additional context**

I would like to work on this.

**Comment by kelson42** [comment]
@mhutti1 @Aditya-Sood @abdulwd @macgills We might need here a bit of vision about next steps (I have remarked as well that I have no way to see all notes!).

**Comment by yashk2000** [comment]
I would like to work on a feature to allow a user to view all notes too.

**Comment by macgills** [comment]
Well this feature sounds valuable anyhow

**Comment by Aditya-Sood** [comment]
@kelson42 @macgills  if the user deletes a particular zim, they aren't able to view the associated notes in-app anymore. Deletion of these notes can only be done either through 'Clear All Notes' or manually.

If that's not okay, we can add the global notes viewer first - that will allow the use of individual note deletion for the above type of notes

**Comment by yashk2000** [comment]
@Aditya-Sood my issue is different. I aim to provide a delete note option for existing notes that can be viewed if they already exist. 

> if the user deletes a particular zim, they aren't able to view the associated notes in-app anymore. Deletion of these notes can only be done either through 'Clear All Notes' or manually.

This is a different issue. 

> we can add the global notes viewer first - that will allow the use of individual note deletion for the above type of notes

I'll be happy to work on a global notes viewer. Since you made the current notes system, could you please give some guidance on what will be the best way to proceed with a global notes viewer?

**Comment by Aditya-Sood** [comment]
@yashk2000 that scenario is relevant because how you've proposed to implement individual note deletion won't work for it, unless there's a global notes viewer

The implementation is pretty straightforward - there's a 'Notes' directory in the Kiwix folder with all the notes arranged by zim. The viewer activity would just have to scan it for all the subdirectories with notes from different zims
(A more comprehensive discussion on implementation is best suited to a new issue)

**Comment by yashk2000** [comment]
@Aditya-Sood I understood that my solution will not work for the issue you mentioned, for that we need to make a global notes viewer and provide the delete option there too. 
But I thought of providing a delete option for the available notes in the add notes dialog too for the convenience of the user. 

> The implementation is pretty straightforward - there's a 'Notes' directory in the Kiwix folder with all the notes arranged by zim. The viewer activity would just have to scan it for all the subdirectories with notes from different zims
(A more comprehensive discussion on implementation is best suited to a new issue)

**Comment by yashk2000** [comment]
@Aditya-Sood thanks for idea.
I understood that my solution will not work for the issue you mentioned, for that we need to make a global notes viewer and provide the delete option there too. But I thought of providing a delete option for the available notes in the add notes dialog too for the convenience of the user. The user doesn't know about the issue that notes from a deleted zim are still present in memory. Users will only be concerned about what they can actually see. 

> The implementation is pretty straightforward - there's a 'Notes' directory in the Kiwix folder with all the notes arranged by zim. The viewer activity would just have to scan it for all the subdirectories with notes from different zims
(A more comprehensive discussion on implementation is best suited to a new issue)

@kelson42 @macgills should we make an issue for this and discuss about the implementation?
